[android] Add sdk45 versioned code

Kudo opened the pull request:
# Why

for sdk 45

# How

based on https://github.com/expo/expo/pull/17116 and do`et add-sdk -p android -s 45.0.0`

# Test Plan

versioned android expo go + sdk45 NCL smoke test

# Checklist

- n/a Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

Kudo commented:
it looks like the prebuilt reanimated libs is debug version, i'll take a look tomorrow.

Kudo committed:
et add-sdk -p android -s 45.0.0

Kudo committed:
spotlessApply

tsapeta reviewed (approved):
Looks good, but let's wait for some QA fixes that may come up later today/tomorrow 😉

Kudo committed:
re-add sdk versioned code

- rebase from latest main(ac975d4d97ed86e4fc66532c7c2bbf78c04878f2)
- cherry-pick https://github.com/expo/expo/pull/17142
